lib/class_extensions.rb

```diff
@@ -0,0 +1,94 @@
+class Array
+  def my_uniq
+    arr = []
+    self.each { |num| arr << num if !arr.include?(num) }
+    p arr
+  end
+
+  def two_sum
+    arr = []
+    (0..self.length).each do |idx|
+      second_pos = idx + 1
+      (second_pos..self.length-1).each do |second_idx|
+        if self[idx] + self[second_idx] == 0
+          arr << [idx, second_idx]
+        end
+      end
+    end
+
+    p arr
+  end
+
+  def median
+    sorted = self.sort
+    if sorted.count.odd?
+      sorted[(sorted.length - 1) / 2]
```

Comment:
Well, if the array count is an odd number, minusing one gives it an even number, so it should always come out to a wholesome number.